x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java

```diff
@@ -197,11 +208,34 @@ public static void copyToCacheFileAligned(
             final int remainder = buf.position() % PAGE_SIZE;
             final int adjustment = remainder == 0 ? 0 : PAGE_SIZE - remainder;
             buf.position(buf.position() + adjustment);
+            final long writeStartTime = System.nanoTime();
             bytesCopied += positionalWrite(fc, fileChannelPos + bytesCopied, buf);
+            totalWriteTimeNanos += System.nanoTime() - writeStartTime;
             final int adjustedBytesCopied = bytesCopied - adjustment; // adjust to not break RangeFileTracker
             assert adjustedBytesCopied == length : adjustedBytesCopied + " vs " + length;
             progressUpdater.accept(adjustedBytesCopied);
         }
+        long elapsedTimeNanos = System.nanoTime() - copyStartTime;
+        metricsConsumer.handleCopyMetrics(totalBytesRead, totalReadTimeNanos, bytesCopied, totalWriteTimeNanos, elapsedTimeNanos);
```

Comment:
You could probably get pretty close to elapsed time by adding read time and write time because they happen sequentially, but having a specific measure means we include things like progress updater callbacks and everything else things that _probably_ make negligible contribution to the overall time, but might also become an issue one day? (also perhaps if someone adds something else that turns out to be expensive in here at some point in the future)